draft-yasskin-dispatch-web-packaging.md

```diff
@@ -0,0 +1,650 @@
+tail = (
+  ; Total number of bytes in the package.
+  length: uint,
+  ; 🌐📦 in UTF-8.
+  magic2: h'F0 9F 8C 90 F0 9F 93 A6',
+)
+~~~~~
+
+If the bytes don't match this group or these two CBOR items don't occupy exactly
+18 bytes, parsing MUST fail.
+
+Otherwise, continue as if the byte `length` bytes before the end of the string
+were the beginning of the package, and the parser were
+a [from the beginning](#from-start) parser.
+
+### From the beginning {#from-start}
+
+If the first 10 bytes of the package are not "85 48 F0 9F 8C 90 F0 9F 93 A6"
+(the CBOR encoding of the 5-item array header and 8-byte bytestring header,
+followed by 🌐📦 in UTF-8), parsing MUST fail.
+
+Parse one CBOR item starting at the 11th byte of the package. If this does not
+match the CDDL
+
+~~~~~ cddl
+section-offsets = { * tstr => uint },
+~~~~~
+
+or it is not a Canonical CBOR item (Section 3.9 of {{CBOR}}), parsing MUST fail.
+
+Let *sections-start* be the offset of the byte after the `section-offsets` item.
+For example, if `section-offsets` were 52 bytes long, *sections-start* would be
+63.
+
+This specification defines two section names: "indexed-content" and "manifest".
+
+If `section-offsets`\["indexed-content"] is not present, parsing MUST fail.
+
+The parser MUST ignore unknown keys in the `section-offsets` map because new
+sections may be defined in future specifications.
+[^ednote-critical-sections]{:source="jyasskin"}
+
+[^ednote-critical-sections]: Do we need to mark critical section names?
+
+Let *index* be the result of parsing the bytes starting at offset
+*sections-start* + `section-offsets`\["indexed-content"] using the instructions in
+{{index}}.
+
+If `section-offsets`\["manifest"] is present, let *manifest* be the
+result of parsing the bytes starting at offset *sections-start* +
+`section-offsets`\["manifest"] using the instructions in {{manifest}}.
+
+The parser MAY return a semantic package consisting of *index*, and,
+if initialized, *manifest*.
+
+To parse each resource described within *index*, the parser MUST follow the
+instructions in {{resource}}.
+
+## Parsing the index {#index}
+
+The main content of a package is an index of HTTP requests pointing to HTTP
+responses. These request/response pairs hold the manifests of sub-packages and
+the resources in the package and all of its sub-packages. Both the requests and
+responses can appear in any order, usually chosen to optimize loading while the
+package is streamed.
+
+To parse the index, starting at offset *index-start*, the parser MUST do the
+following:
+
+If the byte at *index-start* is not 0x82 (the {{CBOR}} header for a 2-element
+array), the parser MUST fail.
+
+Load a CBOR item starting at *index-start* + 1 as the `index` array in the following CDDL:
+
+~~~~~ cddl
+index = [* [resource-key: http-headers,
+            offset: uint,
+            ? length: uint] ]
+
+; http-headers is a byte string in HPACK format (RFC7541).
+; The dynamic table begins empty for each instance of
```

Comment:
Yeah ... How about I delete this comment about the dynamic table and rely on the actual parsing described below.